app/views/orders/index.html.erb

```diff
@@ -1,11 +1,38 @@
-<table class="table align-middle my-5">
+<%= stylesheet_link_tag "pagination/pagination", "data-turbo-track": "reload" %>
+
+<h1>Your orders</h1>
+
+<div class="row mb-2">
+  <div class="col">
+    <%= search_form_for @q, url: orders_path do |f| %>
+      <%= f.search_field :number_cont, placeholder: "Search phrase..." %>
+      <%= f.submit name: '', class: "btn btn-outline-secondary btn-sm mb-1 text-dark" %>
+    <% end %>
+  </div>
+  <div class="col">
+    <%= form_tag orders_path, :method => "get" do %>
+      <span>Orders per page:
+        <%= select_tag :per_page, options_for_select([5, 10, 20, 30], selected: params[:per_page] || 30), :class=>"select_box", :onchange => "this.form.submit()" %>
+    <% end %>
+    </span>
+
+  </div>
+</div>
+
+<div class="apple_pagination">
+  <div class="page_info">
+    <%= page_entries_info @orders %>
+  </div>
+</div>
+
+<table class="table align-middle">
   <thead>
   <tr>
-    <th scope="col">Order Number</th>
-    <th scope="col">Order Date</th>
-    <th scope="col">Quantity</th>
-    <th scope="col">Cost</th>
-    <th scope="col">Status</th>
+    <th scope="col"><%= sort_link(@q, :number, "Order Number",{}, { class: "link-dark" }) %></th>
+    <th scope="col"><%= sort_link(@q, :created_at, "Order Date",{}, { class: "link-dark" }) %></th>
+    <th scope="col"><%= sort_link(@q, :item_quantity, "Quantity",{}, { class: "link-dark" }) %></th>
+    <th scope="col"><%= sort_link(@q, :items_total, "Cost",{}, { class: "link-dark" }) %></th>
+    <th scope="col"><%= sort_link(@q, :status, "Status",{}, { class: "link-dark" }) %></th>
```

Comment:
The name of `@q` variable should be changed.

Reply:
I recieve in the console: ``Passing two trailing hashes to `sort_link` is deprecated, merge the trailing hashes into a single one. ``